lib/spring/application_manager.rb

```diff
@@ -32,7 +32,11 @@ def restart
     end
 
     def alive?
-      @pid
+      return false if @pid.nil?
+      Process.getpgid(@pid)
```

Comment:
This seems to be a reasonable way of determining whether a process with a certain pid exists.